codex/blockexchange/engine/advertiser.nim

```diff
@@ -70,20 +72,26 @@ proc advertiseBlock(b: Advertiser, cid: Cid) {.async.} =
     await b.addCidToQueue(cid)
     await b.addCidToQueue(manifest.treeCid)
 
-proc advertiseLocalStoreLoop(b: Advertiser) {.async.} =
+proc advertiseLocalStoreLoop(b: Advertiser) {.async: (raises: []).} =
   while b.advertiserRunning:
-    if cids =? await b.localStore.listBlocks(blockType = BlockType.Manifest):
-      trace "Advertiser begins iterating blocks..."
-      for c in cids:
-        if cid =? await c:
-          await b.advertiseBlock(cid)
-      trace "Advertiser iterating blocks finished."
+    try:
+      if cids =? await b.localStore.listBlocks(blockType = BlockType.Manifest):
+        trace "Advertiser begins iterating blocks..."
+        for c in cids:
+          if cid =? await c:
+            await b.advertiseBlock(cid)
+        trace "Advertiser iterating blocks finished."
+
+      await sleepAsync(b.advertiseLocalStoreLoopSleep)
 
-    await sleepAsync(b.advertiseLocalStoreLoopSleep)
+    except CancelledError:
```

Comment:
Good catch. Btw, we should probably plan to drop the v3 compat mode from Chronos as it'll then start warning about those all over.